py-polars/polars/utils/lambda_parser.py

```diff
@@ -0,0 +1,257 @@
+"""Try to suggest faster alternatives to simple lambda expressions used in `apply`."""
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING, Any, Callable
+
+from polars.exceptions import PolarsInefficientApplyWarning
+from polars.utils.various import find_stacklevel
+
+if TYPE_CHECKING:
+    import ast
+
+
+def _process_operand(
+    operand: ast.AST, arg: str, names: list[str], level: int, is_expr: bool
+) -> str | None:
+    """
+    Process an operand of a binary operation.
+
+    Parameters
+    ----------
+    operand
+        The operand to process.
+    arg
+        The argument name (e.g. if the lambda is `lambda x: x + 1`, then `arg` is `x`).
+    names
+        The column names (or root names) of the original DataFrame (or Expr).
+    level
+        How deep we are in the AST.
+    is_expr
+        Whether the original object was a DataFrame or an Expr.
+
+
+    Returns
+    -------
+    str | None
+        Return a replacement expression for the `apply` (if found).
+    """
+    import ast
+
+    if isinstance(operand, ast.Subscript):
+        return _process_subscript(operand, arg, names)
+    elif isinstance(operand, ast.BinOp):
+        ret = _process_binop(operand, arg, names, level=level, is_expr=is_expr)
+        if ret is None:
+            return None
+        return f"({ret})"
+    elif isinstance(operand, ast.Constant):
+        return operand.value
+    elif isinstance(operand, ast.Name):
+        if len(names) == 1 and operand.id == arg and is_expr:
+            return f'pl.col("{names[0]}")'
+        return None
+    return None
+
+
+def _process_operator(op: ast.operator) -> str | None:
+    """Return the string representation of some simple operators."""
+    import ast
+
+    if isinstance(op, ast.Add):
+        return "+"
+    elif isinstance(op, ast.Mult):
```

Comment:
How about a dict?
```python
ast_to_string = {ast.Add: "+", ast.Mult: "*",...}
return ast_to_string.get(op)
```

Reply:
thanks - tried this but it doesn't work for me
```python
(Pdb) ast_to_string
{<class 'ast.Add'>: '+', <class 'ast.Mult'>: '*'}
(Pdb) p op
<ast.Add object at 0x7fc3474a39a0>
(Pdb) ast_to_string.get(op)
```

Could use `ast_to_string.get(type(op))`, but I think then it would risk not catching subclasses?

Reply:
List of tuples
```python
ast_to_string = [(ast.Add, "+")]
for ast_op, str_op in ast_to_string:
    if isinstance(op, ast_op):
        return str_op
return None
```

Reply:
Ah, `isinstance`, sorry, that may be tricky indeed. List of tuples seems better indeed.